### PR TITLE
NTUEEPLUS 256. Upload interview draft

### PR DESCRIPTION
**Pull request by Alanhsiu**
### What is this PR for?
create a function of uploading column content

### What type of PR is it?
[Bug Fix | Improvement | Feature | Documentation | Hot Fix | Refactoring]
Feature

### Todos
* [ ] - "body" is not well completed
* [ ] - schema is still needed to be discussed

### What is the Jira issue?
NTUEEPLUS-256. upload interview draft

### Screenshots (if appropriate)
image
image
image
image

### CODE REVIEW

File: backend/routes/srcs/in/career/addRecruitment.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ const asyncHandler = require('express-async-handler')
  */
 const addRecru = async (req, res) => {
   const account = req.session.loginAccount
+  console.log(req.body)
```
Comment (vwvwMM) [inline backend/routes/srcs/in/career/addRecruitment.js:32]: you can delete this console.log

File: backend/routes/srcs/in/column/addColumn.js [hunk 2]
```diff
@@ -10,26 +10,26 @@ const asyncHandler = require('express-async-handler')
  * @apiGroup In/column
  * @apiDescription 管理員新增文章
  *
- * @apiParam {String[]} title 文章標題
+ * @apiParam {String[]} title 文章標題 //要不要刪掉？而且應該沒有[]
  *    (xxxx 級 xxx (公司名稱與職位))(這邊看要不要和name,experience合併?)
- * @apiParam {String} id 文章的編號
+ * @apiParam {String} id 文章的編號 //一個月超過兩次就要改編號
```
Comment (vwvwMM) [inline backend/routes/srcs/in/column/addColumn.js:15]: then you can change it to yymmdd

File: backend/routes/srcs/in/column/addColumn.js [hunk 3]
```diff
@@ -10,26 +10,26 @@ const asyncHandler = require('express-async-handler')
...
  * @apiParam {Object[]} body.body
- * @apiParam {String} body.body.bigtitle (一、標題，二、求學階段...)
+ * @apiParam {String[]} body.body.bigtitle (一、標題，二、求學階段...)
  * @apiParam {Object[]} body.body.bigsections
  * @apiParam {String} body.body.bigsections.subtitle 子標題
  * @apiParam {String} body.body.bigsections.subsection (文章內容)
- * @apiParam {String[]} annotation.annotation 參與全人員
- * @apiParam {String[]} annotation.annotation.job 工作
- * @apiParam {String[]} annotation.annotation.contributer 人員
+ * @apiParam {String[]} annotation.annotation 參與全人員 //應該是Object[]
```
Comment (vwvwMM) [inline backend/routes/srcs/in/column/addColumn.js:27]: then why don't you change it to Object[] ?

File: backend/routes/srcs/in/column/addColumn.js [hunk 4]
```diff
@@ -67,7 +67,22 @@ const asyncHandler = require('express-async-handler')
...
+  const annotation = JSON.parse(req.body.annotation)
+  const anno = JSON.parse(req.body.anno)
+  const exp = JSON.parse(req.body.exp)
+  const edu = JSON.parse(req.body.edu)
+  const intro = JSON.parse(req.body.intro)
+
+  console.log(req.body)
+  console.log(top)
+  console.log(body)
+  console.log(annotation)
```
Comment (vwvwMM) [inline backend/routes/srcs/in/column/addColumn.js:84]: you can delete these console.log

File: backend/routes/srcs/in/column/addColumn.js [hunk 5]
```diff
@@ -79,17 +94,17 @@ const addCol = async (req, res) => {
...
+  //   field: ['top', 'body', 'annotation'],
+  //   type: 'object',
+  // },
   {
     filename: 'optional',
     field: ['date'],
     type: 'string',
   },
-  { filename: 'optional', field: ['anno', 'title', 'exp', 'edu', 'intro'], type: 'array' },
+  // { filename: 'optional', field: ['anno', 'title', 'exp', 'edu', 'intro'], type: 'array' },
```
Comment (vwvwMM) [inline backend/routes/srcs/in/column/addColumn.js:107]: you can delete these comments

File: client/src/_navAuth.js [hunk 6]
```diff
@@ -22,6 +22,13 @@ const _navAuth = [
     to: '/auth/register',
     icon: <CIcon icon="sidebar_icon" name="sidebar_icon" customClassName="nav-icon" />,
   },
+  {
+    _component: 'CNavItem',
+    as: NavLink,
+    anchor: 'COLUMN',
+    to: '/auth/column',
```
Comment (vwvwMM) [inline client/src/_navAuth.js:29]: I think "/auth/addColumn" is better

File: client/src/views/in/column/Title.js [hunk 7]
```diff
@@ -18,11 +18,12 @@ const Title = ({ data }) => {
       <div className="row banner">
         <div className="banner-text">
           <h1 className="responsive-headline">{name}</h1>
-          {exp.map((e) => (
+          {/* {exp.map((e) => (
```
Comment (vwvwMM) [inline client/src/views/in/column/Title.js:21]: you can delete this commment

File: client/src/views/in/recruitment/RecruitmentForm.js [hunk 8]
```diff
@@ -159,6 +159,11 @@ const CareerForm = ({ data }) => {
     const config = {
       headers: { 'content-type': 'multipart/form-data' },
     }
+    console.log(data.get('title'))
+    console.log(data.get('experience[]'))
+    console.log(data.getAll('experience[]'))
+    console.log(typeof data)
+    // console.log(data[experience])
```
Comment (vwvwMM) [inline client/src/views/in/recruitment/RecruitmentForm.js:166]: you can delete these console.log

File: package.json [hunk 9]
```diff
@@ -23,7 +23,9 @@
     "dotenv": "^10.0.0",
     "express": "^4.17.1",
     "express-session": "^1.17.2",
-    "prettier": "^2.3.2"
+    "prettier": "^2.3.2",
+    "yar": "^9.1.0",
+    "yarn": "^1.22.19"
```
Comment (vwvwMM) [inline package.json:28]: what the hell is this?

File: client/src/views/auth/column/index.js [hunk 10]
```diff
@@ -0,0 +1,74 @@
+/* eslint-disable prettier/prettier */
+import ColumnForm from './ColumnForm'
+import { selectCareer, setKeywords, clearKeywords } from '../../../slices/careerSlice'
+import eesa from '../../../assets/images/eesa-icon.png'
+import Recruitment_image from '../../../assets/images/Recruitment.png'
+import Recommendation_image from '../../../assets/images/Recommendation.png'
```
Comment (vwvwMM) [inline client/src/views/auth/column/index.js:6]: why do you import these two pngs?

File: client/src/views/auth/column/index.js [hunk 11]
```diff
@@ -0,0 +1,74 @@
+/* eslint-disable prettier/prettier */
+import ColumnForm from './ColumnForm'
+import { selectCareer, setKeywords, clearKeywords } from '../../../slices/careerSlice'
```
Comment (vwvwMM) [inline client/src/views/auth/column/index.js:3]: why do you import components in careerSlice?

File: client/src/views/auth/index.js [hunk 12]
```diff
@@ -1,6 +1,8 @@
 /* eslint-disable prettier/prettier */
 import React from 'react'
 const AuthMatching = React.lazy(() => import('./matching'))
-const AuthRegister=React.lazy(() => import('./register'))
-const AuthAnnounce=React.lazy(() => import('./announce'))
-export { AuthMatching, AuthRegister,AuthAnnounce }
+const AuthRegister = React.lazy(() => import('./register'))
+const AuthAnnounce = React.lazy(() => import('./announce'))
+const AuthColumn = React.lazy(() => import('./column'))
```
Comment (vwvwMM) [inline client/src/views/auth/index.js:6]: what's different between AuthColumn and AddColumn?

File: client/src/views/auth/column/index.js (hunk 10 above)
Comment (Alanhsiu) [reply client/src/views/auth/column/index.js:6]: to act as the default image

File: client/src/views/auth/column/index.js (hunk 11 above)
Comment (Alanhsiu) [reply client/src/views/auth/column/index.js:3]: those r useful

**Review by vwvwMM — APPROVED** [review]
LGTM

**Pull request merged**